cranelift-codegen/meta/src/isa/x86/encodings.rs

```diff
@@ -1820,23 +1826,26 @@ pub(crate) fn define(
     }
 
     // SIMD insertlane
-    let mut x86_pinsr_mapping: HashMap<u64, (Vec<u8>, Option<SettingPredicateNumber>)> =
-        HashMap::new();
-    x86_pinsr_mapping.insert(8, (vec![0x66, 0x0f, 0x3a, 0x20], Some(use_sse41_simd))); // PINSRB
-    x86_pinsr_mapping.insert(16, (vec![0x66, 0x0f, 0xc4], None)); // PINSRW from SSE2
-    x86_pinsr_mapping.insert(32, (vec![0x66, 0x0f, 0x3a, 0x22], Some(use_sse41_simd))); // PINSRD
-    x86_pinsr_mapping.insert(64, (vec![0x66, 0x0f, 0x3a, 0x22], Some(use_sse41_simd))); // PINSRQ, only x86_64
-
-    for ty in ValueType::all_lane_types().filter(allowed_simd_type) {
-        if let Some((opcode, isap)) = x86_pinsr_mapping.get(&ty.lane_bits()) {
+    for (lane_bits, opcode, isap) in [
```

Comment:
Sorry, this was a bad call from me for insertlane and extractlane; this way of doing it implies to clone the content that's iterated over for no good reason (it's consumed once and for all), plus we now iterate on the all_lane_types() many more times. For these two, let's keep the previous way.